### PR TITLE
Improvement/example page

### LINKED ISSUE #32: Fix flow errors
After the upgrade of flow in #31 there are some new type checking errors, which need to be fixed.

### PR DESCRIPTION
**Pull request by frontendphil**
Fixes #32

I've added a small example application to showcase how the different `substyle` flavors impact the generated HTML. Also, I've fixed the typing errors that have been reported by `flow`. I'm not entirely confident the types are correct but this can be fixed in another PR. 

Note that I've also updated the build setup on the packages to produce `es6` compatible versions next to the regular `es5` build. Also, I've added `flow` output so that the type checker can actually work with this library. My guess is that some more work needs to be done, but you can only tell when we do a pre-release and actually see what happens in other libs. 

Next steps:
- Add Circle CI setup and create a pipeline for QA
- Maybe publish example app automatically using `gh-pages`
- Integration test with libraries that use substyle
- Update `README.md` with finalized new setup instructions

### CODE REVIEW

**Comment by jfschwarz** [comment]
great stuff!

File: packages/example-page/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "example-page",
+  "version": "1.0.0",
+  "description": "Example page highlighting the different flavours of substyle",
+  "main": "src/index.js",
+  "author": "Philipp Giese",
+  "license": "MIT",
+  "private": false,
```
Comment (jfschwarz) [inline packages/example-page/package.json]: Should be `true`, no? Do we want to publish this package to npm?

Comment (frontendphil) [reply packages/example-page/package.json]: Absolutely

File: packages/substyle/src/defaultStyle.js [hunk 2]
```diff
@@ -106,14 +115,6 @@ const createDefaultStyle = (
       if (this.memoizedEnhance !== enhance) {
         this.memoizedEnhance = enhance
         this.enhancedWrappedComponent = enhance(WrappedComponent)
-
-        // eslint-disable-next-line react/forbid-foreign-prop-types
-        if (this.enhancedWrappedComponent.propTypes) {
```
Comment (jfschwarz) [inline packages/substyle/src/defaultStyle.js]: why did you remove this? a test fails because of this.

**Pull request merged**